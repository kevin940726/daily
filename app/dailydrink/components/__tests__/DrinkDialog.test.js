const { html } = require('../../../slack-block-kit');
const DrinkDialog = require('../DrinkDialog');

test('it renders <DrinkDialog />', () => {
  const block = html`
    <${DrinkDialog} state=${{ foo: 'bar' }} />
  `;

  expect(block).toMatchSnapshot();
});

test('it renders <DrinkDialog /> with editing', () => {
  const block = html`
    <${DrinkDialog}
      state=${{ foo: 'bar' }}
      isEdit
      fields=${{
        store: {
          storeName: 'storeName',
          imageURL: 'imageURL',
        },
        title: 'title',
      }}
    />
  `;

  expect(block).toMatchSnapshot();
});
