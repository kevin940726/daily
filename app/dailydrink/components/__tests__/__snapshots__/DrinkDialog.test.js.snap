// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders <DrinkDialog /> 1`] = `
Object {
  "callback_id": "create-new-order-callback-id",
  "elements": Array [
    Object {
      "data_source": "external",
      "hint": "Cannot find the store you are looking for? Try submit a new store via \`/dailydrink submit\`",
      "label": "Store",
      "name": "select-store",
      "selected_options": undefined,
      "type": "select",
    },
    Object {
      "hint": "Provide estimated due time if needed.",
      "label": "Title",
      "name": "title",
      "optional": true,
      "type": "text",
      "value": undefined,
    },
  ],
  "notify_on_cancel": false,
  "state": "{\\"foo\\":\\"bar\\"}",
  "submit_label": "Start",
  "title": "Start a new order",
}
`;

exports[`it renders <DrinkDialog /> with editing 1`] = `
Object {
  "callback_id": "create-new-order-callback-id",
  "elements": Array [
    Object {
      "data_source": "external",
      "hint": "Cannot find the store you are looking for? Try submit a new store via \`/dailydrink submit\`",
      "label": "Store",
      "name": "select-store",
      "selected_options": Array [
        Object {
          "label": "storeName",
          "value": "storeName",
        },
      ],
      "type": "select",
    },
    Object {
      "hint": "Provide estimated due time if needed.",
      "label": "Title",
      "name": "title",
      "optional": true,
      "type": "text",
      "value": "title",
    },
  ],
  "notify_on_cancel": false,
  "state": "{\\"foo\\":\\"bar\\"}",
  "submit_label": "Save",
  "title": "Edit your order",
}
`;
